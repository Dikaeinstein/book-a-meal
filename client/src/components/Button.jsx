import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
