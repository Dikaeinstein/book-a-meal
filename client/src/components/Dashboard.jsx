import React from 'react';
import OrderHistory from './OrderHistory';

const Dashboard = () => <h2>Dashboard</h2>;

export default Dashboard;
