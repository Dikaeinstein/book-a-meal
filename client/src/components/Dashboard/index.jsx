import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DashboardOrderHistory from './DashboardOrderHistory';
import Footer from '../util/Footer';
import FilterOrders from './FilterOrders';
import TotalOrders from './TotalOrders';
import TotalSales from './TotalSales';
import errorHandler from '../util/errorHandler';


const ConnectedDashboard = ({ fetchOrdersError }) => {
  const DashboardOrderHistoryWithErrorHandling =
    errorHandler(DashboardOrderHistory, 'Error fetching orders');

  return (
    <div>
      <main
        className="dash-main text-center bg-dark-light"
        style={{ minHeight: 'calc(100vh - 151px)' }}
      >
        <h1>Dashboard</h1>
        <section className="card dashboard">
          <TotalOrders />
          <TotalSales />
          <FilterOrders />
          <DashboardOrderHistoryWithErrorHandling
            error={fetchOrdersError}
          />
        </section>
      </main>
      <Footer />
    </div>
  );
};

ConnectedDashboard.propTypes = {
  /* eslint react/require-default-props: 0 */
  fetchOrdersError: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string),
  ]),
};

const mapStateToProps = state => ({
  fetchOrdersError: state.orders.error,
});

const Dashboard = connect(mapStateToProps)(ConnectedDashboard);

export default Dashboard;
