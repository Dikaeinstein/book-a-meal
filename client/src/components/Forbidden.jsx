import React from 'react';
import { NavLink } from 'react-router-dom';
import Footer from './util/Footer';

const Forbidden = () => (
  <main>
    <div
      className="forbidden-overlay"
      style={{
        minHeight: 'calc(100vh - 145px)',
        padding: '1rem',
      }}
    >
      <h6
        className="text-center"
        style={{ paddingTop: '60vh' }}
      >
        Sorry, access to this page is denied.
        Either check the url or <NavLink to="/">go home</NavLink>
      </h6>
    </div>
    <Footer />
  </main>
);

export default Forbidden;
