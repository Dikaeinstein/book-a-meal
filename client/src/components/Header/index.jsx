import React from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Logo from './Logo';
import Navigation from './Navigation';

const Header = ({ className }) => (
  <header className={className}>
    <ToastContainer />
    <Logo
      text="Book-A-Meal"
      className="col-1-4 logo text-left text-dark"
    />
    <div className="toggle-show">&#9776;</div>
    <Navigation />
  </header>
);

Header.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Header;
