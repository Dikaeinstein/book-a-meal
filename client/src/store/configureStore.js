import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
