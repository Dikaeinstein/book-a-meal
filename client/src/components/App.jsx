import React, { Component } from 'react';
import Header from './Header';
import IntroBanner from './IntroBanner';
import MenuCard from './MenuCard';
import Footer from './Footer';

const App = () => (
  <div>
    <Header className="navbar" />
    <IntroBanner />
		<MenuCard />
    <Footer />
  </div>
);

export default App;
