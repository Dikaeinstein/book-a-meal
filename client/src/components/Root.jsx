import React from 'react';
import { Provider } from 'react-redux';
import { hot } from 'react-hot-loader';
import {
  Router,
  Route,
  Switch,
  Redirect,
} from 'react-router-dom';
import PropTypes from 'prop-types';
import history from '../helpers/history';

import Header from './Header';
import ConnectedHomePage from './HomePage';
import SignIn from './SignIn';
import SignUp from './SignUp';
import UserMenu from './UserMenu';
import UserOrderHistory from './UserOrderHistory';
import ConnectedPrivateRoute from './PrivateRoute';
import ConnectedPrivateAdminRoute from './PrivateAdminRoute';
import NotFound from './NotFound';
import Dashboard from './Dashboard';
import Meals from './Meals';
import Menus from './Menus';
import OrderConfirmation from './OrderConfirmation';


export const Root = ({ store }) => (
  <Provider store={store}>
    <Router history={history}>
      <div>
        <Header className="navbar" />
        <Switch>
          <Route exact path="/" component={ConnectedHomePage} />
          <Route path="/signin" component={SignIn} />
          <Route path="/signup" component={SignUp} />
          <Route
            path="/signout"
            render={() => (
              <Redirect to="/" />
            )}
          />
          <ConnectedPrivateRoute path="/user-menu" component={UserMenu} />
          <ConnectedPrivateRoute
            path="/user-order-history"
            component={UserOrderHistory}
          />
          <ConnectedPrivateRoute
            path="/order-confirmation"
            component={OrderConfirmation}
          />
          <ConnectedPrivateAdminRoute path="/dashboard" component={Dashboard} />
          <ConnectedPrivateAdminRoute path="/meals" component={Meals} />
          <ConnectedPrivateAdminRoute path="/menus" component={Menus} />
          <Route component={NotFound} />
        </Switch>
      </div>
    </Router>
  </Provider>
);

Root.propTypes = {
  store: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.func,
  ])).isRequired,
};

export default hot(module)(Root);
