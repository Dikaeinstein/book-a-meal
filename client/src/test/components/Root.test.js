import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import ReactModal from 'react-modal';
import { Root } from '../../components/Root';

describe('<Root />', () => {
  let store;
  beforeAll(() => {
    const mockStore = configureStore();
    store = mockStore({ user: { loggedIn: true } });
  });
  it('should redirect to homepage when "/signout route is matched', () => {
    const history = createMemoryHistory({
      initialEntries: ['/signout'], // The initial URLs in the history stack
    });
    const wrapper = shallow(<Root store={store} history={history} />);
    expect(wrapper.find(Route).get(3).props.render()).toBeDefined();
  });
  it('should render Root', () => {
    ReactModal.setAppElement(document.createElement('div'));
    const tree = shallow(<Root store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
