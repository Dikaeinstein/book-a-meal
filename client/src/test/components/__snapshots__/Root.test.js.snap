// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should render Root 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div>
      <Header
        className="navbar"
      />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/signin"
        />
        <Route
          component={[Function]}
          path="/signup"
        />
        <Route
          path="/signout"
          render={[Function]}
        />
        <Connect(PrivateRoute)
          component={[Function]}
          path="/user-menu"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          path="/user-order-history"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          path="/order-confirmation"
        />
        <Connect(PrivateAdminRoute)
          component={[Function]}
          path="/dashboard"
        />
        <Connect(PrivateAdminRoute)
          component={[Function]}
          path="/meals"
        />
        <Connect(PrivateAdminRoute)
          component={[Function]}
          path="/menus"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
  </Router>
</Provider>
`;
